controller/proxy-injector/webhook.go

```diff
@@ -79,6 +78,13 @@ func Inject(api *k8s.API,
 	}
 	proxyInjectionAdmissionRequests.With(admissionRequestLabels(ownerKind, request.Namespace, report.InjectAnnotationAt, configLabels)).Inc()
 
+	for k, v := range nsAnnotations {
```

Comment:
This approach checks only for ns annotations. We should check for both NS and Workload Annotations.